src/test/java/io/confluent/connect/jdbc/sink/integration/PostgresDatatypeIT.java

```diff
@@ -85,10 +91,96 @@ public void after() throws SQLException {
     log.info("Dropped table");
   }
 
+  /**
+   * Verifies that even when the connector encounters exceptions that would cause a connection
+   * with an invalid transaction, the connector sends only the errant record to the error
+   * reporter and establishes a valid transaction for subsequent correct records to be sent to
+   * the actual database.
+   *
+   * @throws Exception
+   */
+  @Test
+  public void testPrimaryKeyConstraintsSendsToErrorReporter() throws Exception {
+    props.put(ERRORS_TOLERANCE_CONFIG, ToleranceType.ALL.value());
+    props.put(DLQ_TOPIC_NAME_CONFIG, DLQ_TOPIC_NAME);
+    props.put(DLQ_TOPIC_REPLICATION_FACTOR_CONFIG, "1");
+    props.put(MAX_RETRIES, "2");
+
+    createTableWithPrimaryKey();
+    connect.configureConnector("jdbc-sink-connector", props);
+    waitForConnectorToStart("jdbc-sink-connector", 1);
+
+    final Schema schema = SchemaBuilder.struct().name("com.example.Person")
+        .field("firstname", Schema.STRING_SCHEMA)
+        .field("lastname", Schema.STRING_SCHEMA)
+        .build();
+    final Struct struct = new Struct(schema)
+        .put("firstname", "Christina")
+        .put("lastname", "Brams");
+
+    String kafkaValue = new String(jsonConverter.fromConnectData(tableName, schema, struct));
+    connect.kafka().produce(tableName, null, kafkaValue);
+    // Send the same record for a PK collision
+    connect.kafka().produce(tableName, null, kafkaValue);
+
+    // Now, create and send another normal record
+    Struct struct2 = new Struct(schema)
+        .put("firstname", "Brams")
+        .put("lastname", "Christina");
+
+    kafkaValue = new String(jsonConverter.fromConnectData(tableName, schema, struct2));
+    connect.kafka().produce(tableName, null, kafkaValue);
+
+    try {
+      // Consume the number of records produced since this is the upper bound of records that
+      // could be sent to the error reporter.
+      ConsumerRecords<byte[], byte[]> records = connect.kafka().consume(3,
+          CONSUME_MAX_DURATION_MS, DLQ_TOPIC_NAME);
+    } catch (RuntimeException e) {
+      // Check that the amount of records that were actually found by the consumer matches the
+      // number of records expected to be sent to the error reporter.
+      assertEquals("1", e.toString().substring(65, 66));
```

Comment:
This seems brittle if https://github.com/apache/kafka/blob/1a09bac0301a15fe7967e9a0c5bf11d34120561b/connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedKafkaCluster.java#L431 ever gets tweaked.